react-native/react/actions/update.js

```diff
@@ -0,0 +1,130 @@
+/* @flow */
+
+import * as Constants from '../constants/update'
+import engine from '../engine'
+import moment from 'moment'
+
+import {updateUi} from '../constants/types/keybase_v1'
+import type {Update} from '../constants/types/flow-types'
+import type {ShowUpdateAction, RegisterUpdateListenerAction, OnCancelAction, OnSkipAction, OnSnoozeAction, OnUpdateAction, SetAlwaysUpdateAction} from '../constants/update'
+import type {Dispatch} from '../constants/types/flux'
+
+let updateResponse: ?(result: any/* UpdatePromptRes */) => void
+export const SNOOZE_TIME_SECS = 60 * 60 * 24
+
+export function registerUpdateListener (): (dispatch: Dispatch) => void {
+  updateResponse = null
+  return dispatch => {
+    engine.listenOnConnect(() => {
+      engine.rpc('delegateUiCtl.registerUpdateUI', {}, {}, (error, response) => {
+        if (error != null) {
+          console.error('error in registering update ui: ', error)
+        } else {
+          console.log('Registered update ui')
+        }
+      })
+    })
+
+    dispatch(({
+      type: Constants.registerUpdateListener,
+      payload: {started: true}
+    }: RegisterUpdateListenerAction))
+
+    const listeners = updateListenersCreator(dispatch)
+    Object.keys(listeners).forEach(
+      k => engine.listenGeneralIncomingRpc(k, listeners[k])
+    )
+  }
+}
+
+function updateListenersCreator (dispatch: Dispatch) {
+  return {
+    'keybase.1.updateUi.updatePrompt': (payload: {update: Update}, response) => {
+      console.log('Asked for update prompt')
+
+      updateResponse = response
+      const {version, description, type, asset} = payload.update
+
+      const windowTitle = {
+        [updateUi.UpdateType.normal]: 'Keybase Update',
+        [updateUi.UpdateType.bugfix]: 'Keybase Update',
+        [updateUi.UpdateType.critical]: 'Critical Keybase Update'
+      }[type]
+
+      const oldVersion = 'TODO GET THIS from config'
```

Comment:
I left some comments above, but I don't see why you couldn't pass in getState here